src/internal/m365/collection/drive/delta_tree_test.go

```diff
@@ -0,0 +1,452 @@
+			parentID:  odConsts.RootID,
+			id:        "folder",
+			name:      "nameyMcNameFace",
+			expectErr: assert.NoError,
+		},
+		{
+			tname:     "add folder",
+			tree:      treeWithRoot(),
+			parentID:  odConsts.RootID,
+			id:        "folder",
+			name:      "nameyMcNameFace",
+			expectErr: assert.NoError,
+		},
+		{
+			tname:     "add package",
+			tree:      treeWithRoot(),
+			parentID:  odConsts.RootID,
+			id:        "folder",
+			name:      "nameyMcNameFace",
+			isPackage: true,
+			expectErr: assert.NoError,
+		},
+		{
+			tname:     "missing ID",
+			tree:      treeWithRoot(),
+			parentID:  odConsts.RootID,
+			name:      "nameyMcNameFace",
+			isPackage: true,
+			expectErr: assert.Error,
+		},
+		{
+			tname:     "missing name",
+			tree:      treeWithRoot(),
+			parentID:  odConsts.RootID,
+			id:        "folder",
+			isPackage: true,
+			expectErr: assert.Error,
+		},
+		{
+			tname:     "missing parentID",
+			tree:      treeWithRoot(),
+			id:        "folder",
+			name:      "nameyMcNameFace",
+			isPackage: true,
+			expectErr: assert.Error,
+		},
+		{
+			tname: "already tombstoned",
+			tree: &folderyMcFolderFace{
+				tombstones: map[string]path.Elements{
+					"folder": path.NewElements(""),
+				},
+			},
+			id:        "folder",
+			name:      "nameyMcNameFace",
+			isPackage: true,
+			expectErr: assert.NoError,
+		},
+		{
+			tname: "add folder before parent",
+			tree: &folderyMcFolderFace{
+				folderIDToNode: map[string]*nodeyMcNodeFace{},
+			},
+			parentID:  odConsts.RootID,
+			id:        "folder",
+			name:      "nameyMcNameFace",
+			isPackage: true,
+			expectErr: assert.Error,
+		},
+		{
+			tname: "folder already exists",
+			tree: &folderyMcFolderFace{
+				folderIDToNode: map[string]*nodeyMcNodeFace{
+					odConsts.RootID: {
+						id:       odConsts.RootID,
+						name:     odConsts.RootPathDir,
+						children: map[string]*nodeyMcNodeFace{},
+					},
+					"other": {
+						id:       "other",
+						name:     "foobar",
+						children: map[string]*nodeyMcNodeFace{},
+					},
+					"folder": {
+						id:        "folder",
+						name:      "nameyMcNameFace",
+						isPackage: true,
+					},
+				},
+			},
+			parentID:  "other",
+			id:        "folder",
+			name:      "nameyMcNameFace",
+			expectErr: assert.NoError,
+		},
+	}
+	for _, test := range table {
+		suite.Run(test.tname, func() {
+			t := suite.T()
+
+			ctx, flush := tester.NewContext(t)
+			defer flush()
+
+			err := test.tree.SetFolder(
+				ctx,
+				test.parentID,
+				test.id,
+				test.name,
+				test.isPackage)
+			test.expectErr(t, err, clues.ToCore(err))
+
+			if err != nil {
+				return
+			}
+
+			result := test.tree.folderIDToNode[test.id]
+			require.NotNil(t, result)
+			assert.Nil(t, result.prev)
+			assert.Equal(t, test.id, result.id)
+			assert.Equal(t, test.name, result.name)
+			assert.Equal(t, test.isPackage, result.isPackage)
+
+			_, ded := test.tree.tombstones[test.id]
+			assert.False(t, ded)
```

Comment:
nit: minor readability update
```suggestion
			assert.NotContains(t, test.tree.tombstones, test.id)
```